core/dbt/contracts/util.py

```diff
@@ -207,13 +208,45 @@ def get_manifest_schema_version(dct: dict) -> int:
     return int(schema_version.split(".")[-2][-1])
 
 
+# we renamed these properties in v1.3
+# this method allows us to be nice to the early adopters
+def rename_metric_attr(data: dict, raise_deprecation_warning: bool = False) -> dict:
+    if raise_deprecation_warning and "sql" in data.keys() or "type" in data.keys():
+        deprecations.warn("metric-attr-renamed")
+    duplicated_attribute_msg = """
+Cannot define both '{}' (old name) and '{}' (new name).
+These are the same attribute. Remove the first and leave the second.
+"""
+    if "sql" in data.keys():
+        if "expression" in data.keys():
+            raise ValidationError(duplicated_attribute_msg.format("sql", "expression"))
+        else:
+            data["expression"] = data.pop("sql")
+    if "type" in data.keys():
```

Comment:
My testing brought my attention to a small issue that people might run into where they might mismatch aspects of the new with the old. Ex:

> calculation_method: expression

Would it be worthwhile to handle that? In my PR I used the following after line 233:

```
    if data["calculation_method"] == "expression":
        calculation_method = "derived"
        data["calculation_method"] = calculation_method
```